src/components/application_manager/include/application_manager/application_manager_impl.h

```diff
@@ -1416,7 +1417,7 @@ class ApplicationManagerImpl
   // Thread that pumps messages audio pass thru to mobile.
   impl::AudioPassThruQueue audio_pass_thru_messages_;
 
-  HMICapabilities hmi_capabilities_;
+  std::auto_ptr<HMICapabilities> hmi_capabilities_;
```

Comment:
@vlantonov, perhaps. But `ApplicationManagerImpl` class have absolute control over `HMICapabilities` and share it only by reference. So, in my opinion, there no need to use `SharedPtr` in this case.